go.mod

```diff
@@ -1,8 +1,6 @@
 module github.com/aquasecurity/trivy
 
-go 1.22
-
-toolchain go1.22.0
```

Comment:
I'm not sure to be honest. Maybe it's redundant to specify toolchain when the go version is the same? https://go.dev/ref/mod#go-mod-file-toolchain

I run the following:

```shell
$ go version
go version go1.22.0 darwin/arm64
```

If you get rid of the patch version from the go directive within go.mod, I get this error:

```shell
$ go mod tidy
go: downloading go1.22 (darwin/arm64)
go: download go1.22 for darwin/arm64: toolchain not available
```

Running `go mod tidy` on the version in main automatically removes the `toolchain` directive and replaces it with this change.

Reply:
hm... this is strange...
my Go doesn't remove `toolchain` 😄 :
```bash
➜  trivy git:(main) git status
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean
➜  trivy git:(main) go mod tidy
➜  trivy git:(main) go version
go version go1.22.2 darwin/arm64
```

Reply:
Hmm I'm taking a look again to see what's going on. I will upgrade to Go 1.22.2 as you and try again.

Reply:
@DmitriyLewen I don't have a good answer but after updating to go 1.22, go removed the toolchain directive for me https://github.com/aquasecurity/trivy/pull/6476/commits/8372bf7d5795863a361ae34731523ffff3bdf471

I personally think since our go version and toolchain are same, this behavior is normal and expected as specifying the same version for the toolchain and go version are redundant. But I'm not sure why I'm not able to replicate the behavior you see on your end.

Could you try the following?
1. Restore the go version from the main branch into my branch
2. Try running `go mod tidy`

Reply:
Go also applies [MVS](https://research.swtch.com/vgo-mvs) to the Go version. Since `trivy-checks` uses `go 1.22.0` now, Go chooses 1.22.0 rather than 1.22.
https://github.com/aquasecurity/trivy-checks/blob/55f3883e2b99a9a48a31c9db17270ab0af5b38f0/go.mod#L3

I'd suggest removing the patch version in trivy-checks. Otherwise, we need to bump the patch version every time.

```
diff --git a/go.mod b/go.mod
index 2cf4c04..6352ca5 100644
--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/aquasecurity/trivy-checks

-go 1.22.0
+go 1.22

 require (
        github.com/aquasecurity/trivy v0.50.2
```

Reply:
Thanks @knqyf263 for that info. I had a hunch it could be that but didn't get to it. Good to know about MVS!

I've updated trivy-checks repo and also used it within Trivy [here](https://github.com/aquasecurity/trivy/pull/6476/commits/78e0ca28407f3c32c7f0bb2f849029d74d7989a1)